toolkit/tools/internal/rpm/rpm.go

```diff
@@ -514,17 +514,37 @@ func filterCompatibleSpecs(inputSpecPaths []string, defines map[string]string) (
 		return
 	}
 
+	type specArchResult struct {
+		compatible bool
+		path       string
+		err        error
+	}
+	resultsChannel := make(chan specArchResult, len(inputSpecPaths))
```

Comment:
I don't think there was any concrete reason for it, other than it reads a bit better to me since the results are iterated via `for i := 0; i < len(inputSpecPaths); i++ {`. Curious what you mean with the 2nd bit though, I think the go routines will just be left orphaned either way until they complete or the chroot closing forcibly kills the tools they are running via `PermanentlyStopAllProcesses()`.

Reply:
If the channel is big enough for all the responses, then none of the goroutines will get stuck when posting to the channel and will always be able to exit. If the channel has size `0`, then if `filterCompatibleSpecs` exits early, then some of the goroutines may get get stuck trying to post to `resultsChannel` and will stick around until the process exits.

Reply:
Oh, right. That makes sense. (I tend to assume 0 size is "unlimited", I always get that backwards...)